python/bifrost/blocks/guppi_raw.py

```diff
@@ -59,7 +59,7 @@ def on_sequence(self, reader, sourcename):
         df_MHz = bw_MHz / nchan
         f0_MHz = cfreq_MHz - 0.5*(nchan-1)*df_MHz
         # Note: This will be negative if OBSBW is negative, which is correct
-        dt_s   = 1. / df_MHz / 1e6
+        dt_s   = np.abs(1. / df_MHz / 1e6)
```

Comment:
Interesting. Could the problem be that OBSBW being negative really means that OBSFREQ is a negative frequency? I.e., what if we moved the sign of OBSBW to OBSFREQ instead, so the axis scale has a negative starting frequency but a positive step. All axes would then have positive steps and merging becomes no problem.

It would be interesting to check if a negative starting frequency works in the FDMT code; I have a feeling it will. Actually, it looks like it may just require some minor numerical changes to avoid complex numbers appearing in the delay computation (the final result is always real).